### PR TITLE
docs: Improve Developer-Guide and SRIOV doc on arm64

### PR DESCRIPTION
**Pull request by hejianet**
This improves the guest kernel build section of Developer-Guide
on arm64.
This also improve the description of sriov use-case.

Fixes: #299

Signed-off-by: Jia He <justin.he@arm.com>

### CODE REVIEW

File: Developer-Guide.md [hunk 1]
```diff
@@ -342,7 +342,9 @@ $ curl -L https://raw.githubusercontent.com/kata-containers/packaging/master/ker
 $ make ARCH=${kernel_dir} -j$(nproc)
 $ kata_kernel_dir="/usr/share/kata-containers"
 $ kata_vmlinuz="${kata_kernel_dir}/kata-vmlinuz-${kernel_version}.container"
-$ [ $kernel_arch = ppc64le ] && kernel_file="$(realpath ./vmlinux)" || kernel_file="$(realpath arch/${kernel_arch}/boot/bzImage)"
+$ [ $kernel_arch = ppc64le ] && kernel_file="$(realpath ./vmlinux)" \
+|| [ $kernel_arch = aarch64] && kernel_file="$(realpath arch/arm64/boot/Image)" \
```
Comment (jodh-intel) [inline Developer-Guide.md]: A couple of things:

- Can you not fold the lines using `\`'s - github renders long lines with scroll bars so they are not necessary.

- This is incorrect as there is no space before `]`. It should be:
  ```
  || [ $kernel_arch = aarch64 ] && kernel_file="$(realpath arch/arm64/boot/Image)"
  ```

**Comment by raravena80** [comment]
@hejianet ping (from your weekly Kata herder) any status?

**Comment by hejianet** [comment]
Sorry for the late response, will change it as commented by jodh-intel

**Comment by hejianet** [comment]
Updated the patch

**Comment by jodh-intel** [comment]
Hi @hejianet - I'm still seeing the backslashes at the ends of the line so I'm not sure if you pushed the correct branch?

---

By the way, although we have tried to avoid backslashes in the past, it was never documented. But it is now:

- https://github.com/kata-containers/documentation/blob/master/Documentation-Requirements.md#code-blocks

File: use-cases/using-SRIOV-and-kata.md [hunk 2]
```diff
@@ -41,6 +41,9 @@ In order to setup your host for SR-IOV, the following has to be true:
   Controller (NIC). The section [Checking your NIC for SR-IOV](#checking-your-nic-for-sr-iov) describes how to assess if you need to make NIC changes and how to make
   the necessary changes.
 
+Besides, you need to enable the NIC driver in your guest kernel config:
+eg. mlx5 for menallox NIC
+
```
Comment (jodh-intel) [inline use-cases/using-SRIOV-and-kata.md]: I'm not really sure I understand this comment. Also, should that be `Mellanox*` ?

/cc @amshinde, who wrote this document.

**Comment by hejianet** [comment]
Sorry, this is the first time for me to send patch by github. I don't know why I closed this issue unconsciously :- (
And I also notice that a new issue is opened unconsciously #309 
Sorry about that

**Comment by jodh-intel** [comment]
@hejianet - no problem! :smile:

**Pull request closed without merging**